File: EarliestRepeated.java
import java.util.*;

public class EarliestRepeated {

    public static char earliestRepeated(String s) {
        Set<Character> set = new HashSet<>();

        for (int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);

            if (set.contains(ch)) {
                return ch;
            } else {
                set.add(ch);
            }
        }
        return '\0';
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter string: ");
        String s = sc.nextLine();

        char ans = earliestRepeated(s);

        if (ans != '\0') {
            System.out.println("Earliest Repeated character: " + ans);
        } else {
            System.out.println("No Repeated character");
        }

        sc.close();
    }
}

File: EarliestRepeated.c
#include <stdio.h>
#include <string.h>

char earliestRepeated(char s[]) {
    int freq[256] = {0};
    int n = strlen(s);
    
    for (int i = 0; i < n; i++) {
        freq[s[i]]++;

        if (freq[s[i]] == 2) { // Second occurrence
            return s[i];
        }
    }
    return '\0'; // No repeated character
}

void main(){
    char str[100];

    printf("Enter string: ");
    scanf("%s", str);

    char ans = earliestRepeated(str);

    if(ans != '\0') {
        printf("Earliest repeated character: %c\n", ans);
    }
    else {
        printf("No repeated character\n");
    }
}